packages/esm-service-queues-app/src/hooks/useQueueEntries.ts

```diff
@@ -0,0 +1,25 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { type QueueEntry, type QueueEntrySearchCriteria } from '../types';
+import useSWR from 'swr';
+
+export function useQueueEntries(searchCriteria?: QueueEntrySearchCriteria) {
+  const searchParam = new URLSearchParams();
+  for (let [key, value] of Object.entries(searchCriteria)) {
+    searchParam.append(key, value.toString());
+  }
+  searchParam.append('v', 'full');
+
+  const apiUrl = `/ws/rest/v1/queue-entry?` + searchParam.toString();
+  const { data, error, isLoading, isValidating, mutate } = useSWR<{ data: { results: Array<QueueEntry> } }, Error>(
+    apiUrl,
+    openmrsFetch,
+  );
+
+  return {
+    queueEntries: data?.data?.results,
+    isLoading,
+    error,
+    isValidating,
+    mutate,
+  };
```

Comment:
I wonder if there is some sort of utility function that exists (or should) that handles this common paradigm of returning the results of a useSWR operation, but unpacking the results from `data?.data?.results`.  @ibacher ?

Reply:
Still a bit cumbersome, but we can at least destructure the returned object by `useSWR` better by doing `{data, ...rest} = ...`, since we don't really touch anything other than `data`.

Reply:
Yeah, the fact that we always have this nested "data?.data?" everywhere always seemed weird to me, a utility function would be nice (though understanding that we are unlikely to go back and refactor the *numerous* existing places where this is used.

Reply:
Ok... I guess that belongs in `esm-core`

Reply:
It's 11 characters and incredibly easy to understand. Please don't write a utility function for this. It would just obfuscate and create pointless interdependency. Would be incredibly annoying to refactor, and to maintain later, and would save absolutely no work in any imaginable future scenario.

Reply:
I wrote a small tirade and then realized I'd misunderstood what was being suggested. The thing that is being suggested, sure, fine. Is it a good use of your time? Hard to say.

Reply:
> the fact that we always have this nested "data?.data?"

Would it be clearer if this the `useSWR` was something like the below?

```ts
const { data: response, ...rest } = useSWR();

return {
  queueEntries: response?.data?.results,
  ...rest
}
```

The pattern is a result of SWR itself calling the export `data`, but what we're actually returning (in most cases) is a [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response) object that's been enhanced with a `data` property that just stores the decoded JSON body (we have a type for this already called `FetchResponse`).

I'm not sure adding such a utility really helps with, though:

```ts
interface RestSearchResults<T> {
  results?: Array<T>;
}

interface RestSearchResultsResponse<T> extends FetchResponse<RestSearchResults<T>> {}

function unwrapSearchResults(response: RestSearchResultsResponse<T> | undefined | null): Array<T> | undefined {
  return response?.data?.results;
}

// usage

const { data: response, ...rest } = useSWR<RestSearchResultsResponse<QueueEntry>, Error>(
  apiUrl,
  openmrsFetch,
);

return {
  queueEntries: unwrapSearchResults(response),
  ...rest
};
```

Reply:
yeah, thinking this through, probably not worth the effort to fix this, but @ibacher  good (for me at least) to understood the explanation of why it works this way, so this thread provided some value.  The "data.data" always looked like it meant we were doing something wrong, but sounds likethe issue is the convention that useSWR uses to call it's response, "data".

Reply:
I have learned something from this thread. Very helpful